### PR TITLE
fix: removed <PrivateWrapper /> component which contributed to page load slowness

### LINKED ISSUE #2439: Bug: Remove unnecessary <PrivateWrapper /> that is slowing down page loads
I was reviewing @Brandon's work for the sidebar for worskpaces and came across this work, https://github.com/open-sauced/app/pull/1208. It's in the base component for Next.js

```
      <SWRConfig
        value={{
          revalidateOnFocus: false,
          shouldRetryOnError: false,
          fetcher: publicApiFetcher,
          provider: localStorageProvider,
        }}
      >
        <NextNProgress options={{ showSpinner: false }} color="hsla(19, 100%, 50%, 1)" height={4} />
        <Toaster />
        <SessionContextProvider supabaseClient={supabaseClient} initialSession={pageProps.initialSession}>
          <PostHogProvider client={posthog}>
            <PrivateWrapper isPrivateRoute={Component.isPrivateRoute}>
              <TipProvider>
                <div className="grid grid-cols-[1fr,auto]">
                  <AppSideBar />

                  {Component.PageLayout ? (
                    <Component.PageLayout>
                      <Component {...pageProps} />
                    </Component.PageLayout>
                  ) : (
                    <Component {...pageProps} />
                  )}
                </div>
              </TipProvider>
            </PrivateWrapper>
          </PostHogProvider>
        </SessionContextProvider>
      </SWRConfig>
```

It's not necessary as we have middleware that already handles routing based on whether or not someone is logged in. This is contributing to slow page loads.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Removes the `<PrivateWrapper />` component from `_app.tsx`. The component was doing unnecessary checks to see if a user was logged in or not. This is already handled by middleware.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [x] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #2439

## Mobile & Desktop Screenshots/Recordings

N/A

### As a logged out user

1. Navigate to http://localhost:3000/hub/insights
2. You're redirected to http://localhost:3000/feed?redirectedFrom=%2Fhub%2Finsights
3. Navigate to http://localhost:3000/hub/insights/new
4. You're redirected to http://localhost:3000/feed?redirectedFrom=%2Fhub%2Finsights%2Fnew
5. TLDR; any page where you should be logged in should redirect you to the feed.

### As a logged in user

1. Ensure you're logged in.
2. Navigate to http://localhost:3000/hub/insights
3. The page loads
4. Navigate to http://localhost:3000/hub/insights/new
5. The page loads
6. TLDR; any page where you should be logged in loads, assuming you have access to it.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: interfaces/with-page-layout.ts [hunk 1]
```diff
@@ -4,5 +4,4 @@ export type WithPageLayout<Q = {}> = NextPage<Q> & {
   PageLayout?: React.ComponentType<any>;
   SEO?: SEOobject;
   updateSEO?: (SEO: SEOobject) => void;
-  isPrivateRoute?: boolean;
```
Comment (nickytonline) [inline interfaces/with-page-layout.ts:7]: I removed all traces of `isPrivateRoute` which the `<PrivateWrapper />` component relied on.

**Comment by bdougie** [comment]
Wow this is quite the improvement. The normal wait time to open an insight page has sped up significantly.

https://github.com/open-sauced/app/assets/5713670/f3062197-252d-4315-9154-6e2b9d11b8fa

**Review by bdougie — APPROVED** [review]
Looks great

File: layouts/hub.tsx [hunk 2]
```diff
@@ -29,23 +28,6 @@ const HubLayout = ({ children }: { children: React.ReactNode }) => {
     return PATHREGEX.test(path);
   };
 
-  useEffect(() => {
```
Comment (nickytonline) [inline layouts/hub.tsx:32]: This whole `useEffect` is no longer required as the middleware handles whether or not someone is logged in as well as if they're onboarded or not.

File: middleware.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { pathToRegexp } from "path-to-regexp";
 // prettier-ignore
 const pathsToMatch = [
   "/",
-  "/hub/insights/:path*",
+  "/hub/:path*",
```
Comment (nickytonline) [inline middleware.ts:12]: Now anything under `/hub` is handled by middleware, e.g. lists.

**Review by jpmcb — APPROVED** [review]
👏👏👏👏👏👏👏👏👏👏

**Comment by bdougie** [comment]
@nickytonline just noticed client side exception on all pages where I am viewing a public insight page.

https://deploy-preview-2440--oss-insights.netlify.app/pages/bdougie/43/dashboard?range=30

**Comment by nickytonline** [comment]
> @nickytonline just noticed client side exception on all pages where I am viewing a public insight page.
> 
> [deploy-preview-2440--oss-insights.netlify.app/pages/bdougie/43/dashboard?range=30](https://deploy-preview-2440--oss-insights.netlify.app/pages/bdougie/43/dashboard?range=30)

Ahh, looks like some hydration issues. I'll get those sorted. This wasn't an issue before because of the `useEffect` that was in place.

CleanShot 2024-01-08 at 22 23 36

**Comment by nickytonline** [comment]
> > @nickytonline just noticed client side exception on all pages where I am viewing a public insight page.
> > [deploy-preview-2440--oss-insights.netlify.app/pages/bdougie/43/dashboard?range=30](https://deploy-preview-2440--oss-insights.netlify.app/pages/bdougie/43/dashboard?range=30)
> 
> Ahh, looks like some hydration issues. I'll get those sorted. This wasn't an issue before because of the `useEffect` that was in place.
> 

Actually the same errors already occur on beta, so it looks like it's a separate issue. I'm assuming you're getting errors like this on my deploy preview @bdougie?

CleanShot 2024-01-08 at 22 34 36

**Comment by bdougie** [comment]
Correct. Only saw it on the deploy preview. We can proceed with this and look into that hydration thing separately in a new PR considering it's already present on beta.

**Pull request merged**